### PR TITLE
Tp pk

### PR DESCRIPTION
**Pull request by phlax**
Add a tp pk to directory

this prevents `parent__parent__parent` type select_relateds and means a directory can always quickly find its lang/project etc

### CODE REVIEW

File: pootle/apps/pootle_app/models/directory.py [hunk 1]
```diff
@@ -56,6 +56,12 @@ class Directory(models.Model, CachedTreeItem):
     # see migration 0005_case_sensitive_schema.py
     pootle_path = models.CharField(max_length=255, null=False, db_index=True,
                                    unique=True, default='/')
+    tp = models.ForeignKey(
+        'pootle_translationproject.TranslationProject',
+        related_name='dirs',
+        null=True,
+        blank=True,
+        db_index=True)
```
Comment (dwaynebailey) [inline pootle/apps/pootle_app/models/directory.py]: Django 1.10 needs `on_delete` to be explicitly see cf7907914f991180161463732b08b4e6d176591d.

File: pootle/apps/pootle_app/migrations/0012_set_directory_tp.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    TP = apps.get_model("pootle_translationproject.TranslationProject")
+    tps = TP.objects.all()
+    for directory in dirs:
+        if not directory.parent:
+            continue
+        parts = directory.pootle_path.split("/")
+        if len(parts) > 4:
+            tp_path = '/'.join(parts[:3] + [""])
+            try:
+                directory.tp = tps.get(directory__pootle_path=tp_path)
```
Comment (unho) [inline pootle/apps/pootle_app/migrations/0012_set_directory_tp.py]: Why not `directory.tp = tps.get(pootle_path=tp_path)` ?

File: pootle/apps/pootle_app/migrations/0012_set_directory_tp.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            except TP.DoesNotExist:
+                continue
```
Comment (unho) [inline pootle/apps/pootle_app/migrations/0012_set_directory_tp.py]: Maybe warn about this problematic directory?

File: pootle/apps/pootle_app/migrations/0012_set_directory_tp.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+
+def set_directory_tp(apps, schema_editor):
+    dirs = apps.get_model("pootle_app.Directory").objects.all()
+    TP = apps.get_model("pootle_translationproject.TranslationProject")
+    tps = TP.objects.all()
+    for directory in dirs:
```
Comment (unho) [inline pootle/apps/pootle_app/migrations/0012_set_directory_tp.py]: I would have done it the other way: for each tp find all dirs whose pootle_path startswith tp.pootle_path and assign the tp to all of them in a bulk update. Is there any specific reason why you chose to do it this way (try..except needed)?

**Comment by unho** [comment]
Nits aside lgtm

File: pootle/apps/pootle_app/migrations/0012_set_directory_tp.py (hunk 4 above)
Comment (phlax) [inline pootle/apps/pootle_app/migrations/0012_set_directory_tp.py]: that makes much more sense - ill update

**Pull request merged**